monai/networks/utils.py

```diff
@@ -376,12 +376,12 @@ def eval_mode(*nets: nn.Module):
     """
 
     # Get original state of network(s)
-    training = [n for n in nets if n.training]
+    training = [n for n in nets if hasattr(n, "training") and n.training]
 
     try:
         # set to eval mode
         with torch.no_grad():
-            yield [n.eval() for n in nets]
+            yield [n.eval() for n in nets if hasattr(n, "eval")]
     finally:
         # Return required networks to training
         for n in training:
```

Comment:
might be useful to add the check `if hasattr(n, "train")` before calling `n.train()`